coalib/results/SourceRange.py

```diff
@@ -10,7 +10,10 @@ class SourceRange(TextRange):
     @enforce_signature
     def __init__(self,
                  start: SourcePosition,
-                 end: (SourcePosition, None)=None):
```

Comment:
Another thing ...
Position should be a property of SourcePosition - as that makes logical sense, and we should use that (like how the file property is done)